### PR TITLE
Add additional system tests

### PR DESCRIPTION
**Pull request by t-young31**
Adds additional system tests. Due to their large size the input and reference output have been uploaded to [Zenodo](https://zenodo.org/) ([arc_01](https://zenodo.org/record/6838866#.YtFemezMKrM), [arc_02](https://zenodo.org/record/6838977#.YtFepezMKrM), [arc_03](https://zenodo.org/record/6839280#.YtFerezMKrM)), rather than the repository and are downloaded when needed.

***
### TODO

- [x] CI caching

***
### Dependencies

#29

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -46,4 +58,8 @@ jobs:
         shell: bash
         run: |
           export OMP_NUM_THREADS=1
-          py.test ${GITHUB_WORKSPACE}/tdms/tests
+          py.test ${GITHUB_WORKSPACE}/tdms/tests/system/ -s -x
```
Comment (t-young31) [inline .github/workflows/tests.yml:61]: `-s` prints stdout and `-x` fails on the first error

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -21,6 +21,18 @@ jobs:
...
+      - name: Setup cache
+        id: cache-test-data
+        uses: actions/cache@v3
+        with:
+          path: test_data
+          key: test_data
+
+      - name: Copy test data
+        if: steps.cache-test-data.outputs.cache-hit == 'true'
+        run: cp -r test_data/ ${GITHUB_WORKSPACE}/tdms/tests/system/data
```
Comment (giordano) [inline .github/workflows/tests.yml:34]: Why not simply caching `${GITHUB_WORKSPACE}/tdms/tests/system/data`?

File: .gitignore [hunk 3]
```diff
@@ -8,3 +8,4 @@ doc/*.pdf
 **/.pytest_cache/
 **.pdf
 **cmake-build**
+tdms/tests/system/data
```
Comment (giordano) [inline .gitignore:11]:
```suggestion
tdms/tests/system/data

```

File: .github/workflows/tests.yml (hunk 2 above)
Comment (t-young31) [reply .github/workflows/tests.yml:34]: An excellent question... (blindly copying the example might have had something to do with it)

**Comment by giordano** [comment]
Uh, tests now take some time.  But https://github.com/UCL/TDMS/runs/7424344884?check_suite_focus=true#step:17:2
```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

**Pull request merged**